src/nnet3/nnet-simple-component.cc

```diff
@@ -1291,6 +1292,293 @@ void RepeatedAffineComponent::UnVectorize(const VectorBase<BaseFloat> &params) {
                                         bias_params_.Dim()));
 }
 
+BlockAffineComponent::BlockAffineComponent(const BlockAffineComponent &other) :
+  UpdatableComponent(other),
+  linear_params_(other.linear_params_),
+  bias_params_(other.bias_params_),
+  num_blocks_(other.num_blocks_) {}
+
+Component* BlockAffineComponent::Copy() const {
+  BlockAffineComponent *ans = new BlockAffineComponent(*this);
+  return ans;
+}
+
+std::string BlockAffineComponent::Info() const {
+  std::stringstream stream;
+  BaseFloat linear_params_size = static_cast<BaseFloat>(linear_params_.NumRows())
+    * static_cast<BaseFloat>(linear_params_.NumCols());
+  BaseFloat linear_stddev =
+      std::sqrt(TraceMatMat(linear_params_, linear_params_, kTrans) /
+                linear_params_size),
+    bias_stddev = std::sqrt(VecVec(bias_params_, bias_params_) /
+                            bias_params_.Dim());
+
+  stream << Type() << ", input-dim=" << InputDim()
+         << ", output-dim=" << OutputDim()
+         << ", num-blocks=" << num_blocks_
+         << ", linear-params-stddev=" << linear_stddev
+         << ", bias-params-stddev=" << bias_stddev
+         << ", learning-rate=" << LearningRate()
+         << ", is-gradient=" << (is_gradient_ ? "true" : "false");
+  return stream.str();
+}
+
+void BlockAffineComponent::Init(BaseFloat learning_rate, int32 input_dim,
+                                int32 output_dim, int32 num_blocks,
+                                BaseFloat param_stddev, BaseFloat bias_stddev) {
+  KALDI_ASSERT(input_dim > 0 && output_dim > 0 && num_blocks >= 1);
+  KALDI_ASSERT(output_dim % num_blocks == 0 && input_dim % num_blocks == 0);
+  const int32 num_columns_per_block = input_dim / num_blocks;
+
+  UpdatableComponent::Init(learning_rate);
+
+  linear_params_.Resize(output_dim, num_columns_per_block);
+  bias_params_.Resize(output_dim);
+  KALDI_ASSERT(param_stddev >= 0.0 && bias_stddev >= 0.0);
+  linear_params_.SetRandn();
+  linear_params_.Scale(param_stddev);
+  bias_params_.SetRandn();
+  bias_params_.Scale(bias_stddev);
+
+  num_blocks_ = num_blocks;
+}
+
+void BlockAffineComponent::InitFromConfig(ConfigLine *cfl) {
+  int32 input_dim, output_dim, num_blocks;
+  if(!cfl->GetValue("input-dim", &input_dim) ||
+     !cfl->GetValue("output-dim", &output_dim) ||
+     !cfl->GetValue("num-blocks", &num_blocks) || cfl->HasUnusedValues()) {
+    KALDI_ERR << "Invalid initializer for layer of type "
+              << Type() << ": \"" << cfl->WholeLine() << "\"";
+  }
+
+  // optional parameters
+  BaseFloat learning_rate = learning_rate_;
+  cfl->GetValue("learning-rate", &learning_rate);
+  BaseFloat param_stddev = 1.0 / std::sqrt(input_dim),
```

Comment:
I assume you're talking about Glorot intialization. If so, I agree with these changes.